miniapps/tribol/ContactPatchTest.cpp

```diff
@@ -0,0 +1,368 @@
+   mfem::STRUMPACKSolver solver(MPI_COMM_WORLD);
+   solver.SetKrylovSolver(strumpack::KrylovSolver::DIRECT);
+   solver.SetReorderingStrategy(strumpack::ReorderingStrategy::METIS);
+   solver.SetPrintFactorStatistics(true);
+   solver.SetPrintSolveStatistics(true);
+   solver.SetOperator(*A_strumpk);
+#else
+   mfem::MINRESSolver solver(MPI_COMM_WORLD);
+   solver.SetRelTol(1.0e-12);
+   solver.SetMaxIter(2000);
+   solver.SetPrintLevel(3);
+   solver.SetOperator(*A_hyprePar);
+   mfem::HypreDiagScale prec(*A_hyprePar);
+   solver.SetPreconditioner(prec);
+#endif
+   solver.Mult(B_blk, X_blk);
+
+   // Update displacement and coords grid functions
+   auto& displacement_true = X_blk.GetBlock(0);
+   fespace.GetProlongationMatrix()->Mult(displacement_true, displacement);
+   displacement.Neg();
+   coords += displacement;
+
+   // Update the pressure grid function
+   auto& pressure_true = X_blk.GetBlock(1);
+   P_submesh.Mult(pressure_true, pressure);
+
+   // Verify the forces are in equilibrium, i.e. f_int = A*u = -f_contact = -B^T*p
+   // This should be true if the solver converges.
+   mfem::Vector f_int_true(fespace.GetTrueVSize());
+   f_int_true = 0.0;
+   mfem::Vector f_contact_true(f_int_true);
+   A_blk->GetBlock(0, 0).Mult(displacement_true, f_int_true);
+   A_blk->GetBlock(0, 1).Mult(pressure_true, f_contact_true);
+   mfem::Vector resid_true(f_int_true);
+   resid_true += f_contact_true;
+   for (int i{0}; i < ess_tdof_list.Size(); ++i)
+   {
+      resid_true[ess_tdof_list[i]] = 0.0;
+   }
+   auto resid_linf = resid_true.Normlinf();
+   if (mfem::Mpi::Root())
+   {
+      MPI_Reduce(MPI_IN_PLACE, &resid_linf, 1, MPI_DOUBLE, MPI_MAX, 0,
+                 MPI_COMM_WORLD);
+      std::cout << "|| force residual ||_(infty) = " << resid_linf << std::endl;
+   }
+   else
+   {
+      MPI_Reduce(&resid_linf, &resid_linf, 1, MPI_DOUBLE, MPI_MAX, 0, MPI_COMM_WORLD);
+   }
+
+   // Verify the gap is closed by the displacements, i.e. B*u = gap
+   // This should be true if the solver converges.
+   mfem::Vector gap_resid_true(gap_true.Size());
+   gap_resid_true = 0.0;
+   A_blk->GetBlock(1, 0).Mult(displacement_true, gap_resid_true);
+   gap_resid_true -= gap_true;
+   auto gap_resid_linf = gap_resid_true.Normlinf();
+   if (mfem::Mpi::Root())
+   {
+      MPI_Reduce(MPI_IN_PLACE, &gap_resid_linf, 1, MPI_DOUBLE, MPI_MAX, 0,
+                 MPI_COMM_WORLD);
+      std::cout << "|| gap residual ||_(infty) = " << gap_resid_linf << std::endl;
+   }
+   else
+   {
+      MPI_Reduce(&gap_resid_linf, &gap_resid_linf, 1, MPI_DOUBLE, MPI_MAX, 0,
+                 MPI_COMM_WORLD);
+   }
+
+   // Update the Tribol mesh based on deformed configuration
+   // NOTE: This is done to update the contact submesh based on the current
+   // deformed shape of the mesh.
+   tribol::updateMfemParallelDecomposition();
+
+   // Save data in VisIt format
+   mfem::VisItDataCollection visit_vol_dc("ContactPatchTestVolume", &mesh);
```

Comment:
These files have to be deleted with `make clean` in the miniapp make file and currently they are not

Reply:
Thanks for catching this.  It's fixed!